src/components/Widgets/FeeContribution/FeeContribution.tsx

```diff
@@ -0,0 +1,506 @@
+          donation_amount_usd: usdAmount || 0,
+          donation_amount_token: tokenAmount || 0,
+          donation_token_symbol: completedRoute?.toToken?.symbol || '',
+          donation_token_address: completedRoute?.toToken?.address || '',
+          donation_chain: completedRoute?.toChainId || 0,
+        },
+      });
+
+      // Convert to token units with proper precision
+      const amountInTokenUnits = parseUnits(
+        tokenAmount,
+        completedRoute.toToken.decimals,
+      );
+
+      if (completedRoute.toToken.address === DEFAULT_WALLET_ADDRESS) {
+        const nativeTxConfig = createNativeTransactionConfig(
+          feeAddress as `0x${string}`,
+          amountInTokenUnits,
+        );
+        await sendTransaction(nativeTxConfig);
+      } else {
+        // Type assertion to ensure we're using the token transaction config
+        const txConfig = createTokenTransactionConfig(
+          completedRoute.toToken.address as `0x${string}`,
+          feeAddress as `0x${string}`,
+          amountInTokenUnits,
+          completedRoute.toChainId,
+        ) as {
+          address: `0x${string}`;
+          abi: typeof ERC20_TRANSFER_ABI;
+          functionName: 'transfer';
+          args: [`0x${string}`, bigint];
+        };
+
+        await writeContract(txConfig);
+      }
+    } catch (error) {
+      console.error(translations.error.errorSending, error);
+      if (error instanceof Error) {
+        const errorMessage = error.message;
+        if (errorMessage === translations?.error?.amountTooSmall) {
+          console.error(errorMessage);
+        } else if (errorMessage === translations?.error?.noFeeAddress) {
+          console.error(errorMessage);
+        } else if (errorMessage === translations?.error?.invalidTokenPrice) {
+          console.error(errorMessage);
+        }
+      }
+    } finally {
+      setIsSending(false);
+    }
+  };
+
+  const isCustomAmountActive = useMemo(() => {
+    return (
+      !!amount &&
+      !contributionOptions.includes(parseFloat(amount)) &&
+      parseFloat(amount) > 0
+    );
+  }, [amount]);
+
+  if (!showContribution) {
+    return null;
+  }
+
+  return (
+    <ContributionWrapper showContribution={showContribution}>
+      <DrawerWrapper ref={containerRef}>
+        <Drawer
+          open={showContribution}
+          anchor="top"
+          hideBackdrop={true}
+          container={() => containerRef.current}
+          sx={{ position: 'absolute' }}
+          ModalProps={{
+            disablePortal: true,
+            disableEnforceFocus: true,
+            disableAutoFocus: true,
+          }}
+          slotProps={{
+            paper: {
+              sx: {
+                position: 'absolute',
+                bottom: 0,
+                left: 0,
+                right: 0,
+                width: '100%',
+                borderRadius: '24px',
+              },
+            },
+            backdrop: {
```

Comment:
if I understand correctly, you've disabled the backdrop so this might not be necessary